.github/workflows/test.yml

```diff
@@ -20,4 +38,45 @@ jobs:
     steps:
     - uses: seL4/ci-actions/camkes-vm@master
       with:
+        xml: ${{ needs.code.outputs.xml }}
         march: ${{ matrix.march }}
+    - name: Upload images
+      uses: actions/upload-artifact@v3
+      with:
+        name: images-${{ matrix.march }}
+        path: '*-images.tar.gz'
+
+  hw-run:
+    name: Hardware
+    runs-on: ubuntu-latest
+    needs: [build]
```

Comment:
Is there any specific reason why you use the brackets here, but not in line 32 ("` needs: code`")? Seems It's copy/pasted over several repos, so the the original reason might have been that there was more in this array in another repo. We could you the array syntax everywhere for consistency.

Reply:
Both work, but yes always using a list would be more consistent.